rpcs3/rpcs3qt/emu_settings.cpp

```diff
@@ -307,6 +307,86 @@ void emu_settings::EnhanceCheckBox(QCheckBox* checkbox, emu_settings_type type)
 	});
 }
 
+void emu_settings::EnhanceDateTimeEdit(QDateTimeEdit* date_time_edit, emu_settings_type type, const QString& format, bool use_calendar, bool as_offset_from_now)
+{
+	if (!date_time_edit)
+	{
+		cfg_log.fatal("EnhanceDateTimeEdit '%s' was used with an invalid object", cfg_adapter::get_setting_name(type));
+		return;
+	}
+
+	date_time_edit->setDisplayFormat(format);
+	date_time_edit->setCalendarPopup(use_calendar);
+
+	if (as_offset_from_now)
+	{
+		bool ok_def = false;
+		const s64 def = qstr(GetSettingDefault(type)).toLongLong(&ok_def);
+		if (!ok_def)
+		{
+			cfg_log.fatal("EnhanceDateTimeEdit '%s' was used with an invalid emu_settings_type", cfg_adapter::get_setting_name(type));
+			return;
+		}
+
+		bool ok_sel = false;
+		s64 val = qstr(GetSetting(type)).toLongLong(&ok_sel);
+		if (!ok_sel)
+		{
+			cfg_log.error("EnhanceDateTimeEdit '%s' tried to set an invalid value: %d. Setting to default: %d", cfg_adapter::get_setting_name(type), val, def);
+			val = def;
+			m_broken_types.insert(type);
+			SetSetting(type, std::to_string(def));
+		}
+
+		// get the date_time value with the offset applied, and set it to the control
+		const QDateTime date_time = QDateTime::currentDateTime().addSecs(val);
```

Comment:
I thought this was qint64. Is this a valid assignment?

Reply:
qint64 is a typedef for a signed 64 bit integer (our s64).
Given that the EMU core doesn't want to be speaking qint64 terminology, maintaining consistency as s64 in all places apart from these two functions (putting offset into addSecs(), and taking offset out as toDate()) is 'better' in my not so humble opinion.

Reply:
no. what I meant was that I thought addSecs returns qint64. But I was in the wrong line in the docs, so it doesn't matter anyway.

We never specified that you are supposed to use s64. I don't know why you think that.
This is a rule for emucore.

Reply:
QDateTime.addSecs(long long secs) returns QDateTime.